### ISSUE TITLE
Update Django to latest 4.2 LTS

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2750]
Our current testing phase is focused on updating our dependencies. The second largest of which is Django itself (Python has mostly been addressed but is awaiting a Py3.10 update).

Canonical issue for Milestone: https://github.com/rockstor/rockstor-core/milestone/26

From: https://docs.djangoproject.com/en/4.2/faq/install/#what-python-version-can-i-use-with-django

Django version | Python versions
| - | - |
4.1	| 3.8, 3.9, 3.10, 3.11 (added in 4.1.3)
4.2	| 3.8, 3.9, 3.10, 3.11

**Comment by phillxnet** [comment]
Moving to a 4.2.* pinning and `poetry update` gives:
```
lbuildvm:/opt/rockstor # poetry update
Updating dependencies
Resolving dependencies... (1.7s)

Writing lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating django (3.2.23 -> 4.2.7)
```
So no incidental updates as of yet, plus no remaining dependency blockers.

**Comment by phillxnet** [comment]
# WARNINGS

- [ ] /opt/rockstor/.venv/lib/python3.9/site-packages/django/conf/__init__.py:267: RemovedInDjango50Warning: The USE_L10N setting is deprecated. Starting with Django 5.0, localized formatting of data will always be enabled. For example Django will display numbers and dates using the format of the current locale.
 warnings.warn(USE_L10N_DEPRECATED_MSG, RemovedInDjango50Warning)
- [ ] /opt/rockstor/.venv/lib/python3.9/site-packages/django/conf/__init__.py:289: RemovedInDjango51Warning: The STATICFILES_STORAGE setting is deprecated. Use STORAGES instead.
 warnings.warn(STATICFILES_STORAGE_DEPRECATED_MSG, RemovedInDjango51Warning)
- [ ] /opt/rockstor/src/rockstor/scripts/scheduled_tasks/reboot_shutdown.py:24: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/smart_manager/agents/nfsd_calls.py:10: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/smart_manager/views/base_service.py:27: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/smart_manager/views/replication.py:31: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/smart_manager/views/replica_trail.py:20: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/smart_manager/views/replica_share.py:27: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/smart_manager/views/receive_trail.py:20: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/storageadmin/views/share_helpers.py:20: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/storageadmin/views/command.py:46: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc
- [ ] /opt/rockstor/src/rockstor/storageadmin/views/disk_smart.py:45: RemovedInDjango50Warning: The django.utils.timezone.utc alias is deprecated. Please update your code to use datetime.timezone.utc instead.
  from django.utils.timezone import utc

**Comment by phillxnet** [comment]
# TESTS
```
cd /opt/rockstor/src/rockstor/
export DJANGO_SETTINGS_MODULE=settings
poetry run django-admin test -v 2
...
WARNINGS:
?: (staticfiles.W004) The directory '/opt/rockstor/jslibs' in the STATICFILES_DIRS setting does not exist.
...
----------------------------------------------------------------------
Ran 278 tests in 37.773s

OK
```

So we seem to be in fairly good shape on this update front. I will begin to address the warnings spam as this will otherwise hinder our future development.

**Comment by phillxnet** [comment]
## Issue branch (as yet un-plublished):
N.B. remaining issue to-date.
```
lbuildvm:/opt/rockstor # poetry run python -Wa .venv/bin/django-admin test
/opt/rockstor/.venv/lib/python3.9/site-packages/django/conf/__init__.py:267: RemovedInDjango50Warning: The USE_L10N setting is deprecated. Starting with Django 5.0, localized formatting of data will always be enabled. For example Django will display numbers and dates using the format of the current locale.
  warnings.warn(USE_L10N_DEPRECATED_MSG, RemovedInDjango50Warning)
Found 0 test(s).
System check identified no issues (0 silenced).

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK

```

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2752 

Thanks also to @Hooverdan96 for their preliminary investigations re an experimental Django update PR:
[(t) Experimental Django update](https://github.com/rockstor/rockstor-core/pull/2372) some time ago.
